GpsDataAnalyzer/calculator/alignment_algorithms.py

```diff
@@ -237,9 +239,14 @@ def find_best_data_range(primary_time_points_mapping, secondary_time_points_mapp
         end_time = start_time + timedelta(seconds=range_optimization_size-1)  # point at end of current range that was just added
         if end_time not in primary_time_points_mapping or end_time not in secondary_time_points_mapping:
             current_range_skip_count += 1
+        else:
+            overlapping_timestamps.add(end_time)
 
         previous_start = start_time + timedelta(seconds=-1)  # point that was just edged out
-        if previous_start not in primary_time_points_mapping or previous_start not in secondary_time_points_mapping:
+        if previous_start in overlapping_timestamps:
+            # remove from set to keep set size of max range_optimization_size
```

Comment:
Sorry, I'm not sure why we need to keep the set to a specific size since it's not used outside this function. Since we're removing from the start of the range, it's not as if the same point will be encountered again right?

Reply:
Chris wanted me to optimize is so that there weren't 2n calls for checking membership in each dictionary, keeping a fixed size set of the upcoming previous_start values rather than keeping them all was the only way I could think of to do that, but at the end of the day I think time-wise it doesn't really make much of a difference.

Reply:
Yeah, I think it might be premature optimization. It's a bit harder to read, but not that bad, so I think it's fine either way.